src/os/bluestore/BlueStore.cc

```diff
@@ -3533,6 +3533,10 @@ void *BlueStore::MempoolThread::entry()
       next_resize += resize_interval;
     }
 
+    if (ceph_clock_now() - store->get_deferred_last_submitted() >
```

Comment:
Currently we use a very small bluestore_cache_trim_interval to make sure the total memory an osd can consume is under control.  Seems this way we'll try to acquire the deferred_lock very frequently. Not sure if there will be any performance-related concern.

Shall we instead move this block a few lines above, to instead share the autotune_interval setting (which is 5s) and hence slow down the frequency of  calling `get_deferred_last_submitted()`?

@ifed01 @liewegas What do you think?

Reply:
thanks, that make sense.

Reply:
I would refrain from using autotune_interval to avoid potential side effects when we tune (or even disable) it. IMO it's better to wrap get_defferred_last_submitted() call with another 'if' block to reduce the call frequency, i.e. just perform get_defferred_last_submitted() check once a second.

Reply:
@ifed01 yeah,  it's generally not a good idea to scope two different things with one single option.